src/games/strategy/triplea/ui/export/ScreenshotExporter.java

```diff
@@ -0,0 +1,205 @@
+package games.strategy.triplea.ui.export;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.imageio.ImageIO;
+import javax.swing.JFileChooser;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import javax.swing.filechooser.FileFilter;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.framework.ui.SaveGameFileChooser;
+import games.strategy.engine.history.HistoryNode;
+import games.strategy.engine.history.Round;
+import games.strategy.triplea.ui.IUIContext;
+import games.strategy.triplea.ui.MapPanel;
+import games.strategy.triplea.ui.TripleAFrame;
+import games.strategy.triplea.ui.mapdata.MapData;
+import games.strategy.ui.SwingComponents;
+import games.strategy.ui.Util;
+
+/**
+ * Exports a game screenshot.
+ */
+public final class ScreenshotExporter {
+  private final TripleAFrame frame;
+
+  /**
+   * Initializes a new instance of the {@code ScreenshotExporter} class.
+   * 
+   * @param frame The frame associated with the game screenshot to export; must not be {@code null}.
+   */
+  public ScreenshotExporter(final TripleAFrame frame) {
+    this.frame = checkNotNull(frame);
+  }
+
+  /**
+   * Prompts the user for the file to which the screenshot will be saved and saves the screenshot for the specified game
+   * at the specified history step to that file.
+   * 
+   * @param frame The frame associated with the game screenshot to export; must not be {@code null}.
+   * @param gameData The game data; must not be {@code null}.
+   * @param node The history step at which the game screenshot is to be captured; must not be {@code null}.
+   * 
+   * @return A promise that is fulfilled when the save screenshot operation is complete or empty if the user aborted the
+   *         save; never {@code null}.
+   */
+  public static Optional<CompletableFuture<Void>> promptForFileAndRunSave(final TripleAFrame frame,
+      final GameData gameData, final HistoryNode node) {
+    final ScreenshotExporter exporter = new ScreenshotExporter(frame);
+    return exporter.promptForFile()
```

Comment:
Agreed based on the previous comment to change the return value to `void`.  Will adjust as suggested.